### DISCUSSION TITLE
How to use TearingScenarioEngine?

### DISCUSSION THREAD

**Discussion by RickYang1** [InfinyTech3D/SofaUnity#127] [Help - Using SofaUnity]
I am developing continuous tearing simulation, I’ve tested the `TearingScenarioEngine` with following code：
```
int previousVertexCount= Tissue.GetMesh().vertices.Length;	//get tissue's vertex number
while (true)
{
    if (Input.GetKeyDown(KeyCode.Space))
    {
        Vector3[] vertices = Tissue.GetMesh().vertices;
        int currentVertexCount = vertices.Length;
        if (currentVertexCount==previousVertexCount)	//keep looping if nothing happens to the tissue
        {
            continue;
        }
		//find the farthest vertex along the tearing path
        int farthestIndex = FindFarthestPointAlongDirection(vertices, startVertexId.Value, startDirection.Value, previousVertexCount);
        nbFractureMax.Value++;
        Debug.DrawLine(vertices[farthestIndex], vertices[farthestIndex] + Vector3.up * 2, Color.green, 50f); 
		// update tearing path extending info
        startVertexId.Value = farthestIndex;
        startDirection.Value = Quaternion.Euler(0, 10, 0) * startDirection.Value;
		previousVertexCount = currentVertexCount;  // update tissue's vertex number
    }
    yield return new WaitForSeconds(0.02f);
}
```
However, when I pull the tissue or cut it too quickly, Unity crashes：
https://github.com/user-attachments/assets/c07e93b8-301d-4b34-bebd-5c1f9a93fc43
The crash log is as follows:
[Editor.log](https://github.com/user-attachments/files/17040349/Editor.log)
May I ask how to avoid this problem？
In addition, the parameter `startVertexId `of `TearingScenarioEngine `is the index of the midpoint of the tearing path. 
But in my case, I'd like to use the endpoint of last fracture, which is also the starting point of the next fracture, as input parameter,
since there could be no vertex just happen to be at the middle of my expected tearing path.
Which is, according to line 121 of TearingScenarioEngine.inl:
![image](https://github.com/user-attachments/assets/a2b3ec2c-27cf-464c-bdeb-acc8cc6f0a3c)
I'd like to invoke it with Pc and the fracture direction, but not Pa and the direction.
I don't know if this could be done. 
Thanks in advance!

**Comment by epernod** [comment]
Hi,

I understand... This is something that could be developed. I will see if we can add that to the roadmap of next month.
Otherwise feel free to propose some change in the code of the [Tearing plugin](https://github.com/InfinyTech3D/Tearing), we will review your Pull Request and we could add and regenerate a unity compatible version.
Best regards,

**Reply by RickYang1** [reply]
Thanks for your reply. It would be great if these requirements could be added to your roadmap.
Regarding the issues with easy crashes, I still haven’t pinpointed the cause and am unable to make modifications at this time. 
Additionally, I’ve noticed that for smaller models (with a diameter of about 0.01m), the actual tearing path deviates from the preset path. What could be causing this?
Best regards!

**Reply by epernod** [reply]
Hi,

sorry for the late answer. Are you still stuck or did you solved your issue?

Best regards
